pkg/cosign/registryOptions.go

```diff
@@ -0,0 +1,79 @@
+package cosign
```

Comment:
Why not use https://github.com/sigstore/cosign/blob/65eb28af970d133adeefdc6c48d6e9304dd8cc3a/cmd/cosign/cli/options/registry.go#L88?

Reply:
I decided against using it because it is in the CLI package, and since we are trying to remove CLI packages I tried to create our own. There is no package that provides it other than the CLI package. Also the function used here is quite similar to the notary authentication, I am thinking about making a common package that will fetch gcr remote options for both of them

Reply:
Ok. Should we change this now, or wait till we remove other Cosign CLI dependencies?

Reply:
This PR removes all the other CLI dependencies as well